### PR TITLE
[multikueue] Add Path kubeConfig location type

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Add Path location type for MultiKueue cluster KubeConfigs
#### Which issue(s) this PR fixes:

Relates to #693

#### Does this PR introduce a user-facing change?

```release-note
Add Path location type for MultiKueue cluster KubeConfigs
```

### CODE REVIEW

**Comment by trasc** [comment]
/hold

for #1631

File: pkg/controller/admissionchecks/multikueue/multikueuecluster.go [hunk 1]
```diff
@@ -220,6 +221,7 @@ func (c *clustersReconciler) Reconcile(ctx context.Context, req reconcile.Reques
 	}
 	if err != nil {
 		log.Error(err, "reading kubeconfig")
+		c.stopAndRemoveCluster(req.Name)
```
Comment (mimowo) [inline pkg/controller/admissionchecks/multikueue/multikueuecluster.go:224]: This does not seem directly related, is it a drive-by fix to ensure there is no memory leak? Is there a unit test which would fail without it?

**Review by mimowo — COMMENTED** [review]
LGTM overall, one question regarding the not-so-clear change: https://github.com/kubernetes-sigs/kueue/pull/1640/files#r1469263382

**Review by mimowo — COMMENTED** [review]
/lgtm
/assign @alculquicondor @tenzen-y

File: apis/kueue/v1alpha1/multikueue_types.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ const (
 type LocationType string
 
 const (
+	// Location is the path on the disk of kueue-controller-manager.
+	PathLocationType LocationType = "Path"
```
Comment (tenzen-y) [inline apis/kueue/v1alpha1/multikueue_types.go:32]: This `Path` location type is assumed to be the main story that we want to use credentials that existed on the Node. It means that the main use case is mounting credentials by hostPath volume, right?

File: pkg/controller/admissionchecks/multikueue/testdata/worker1KubeConfig
Comment (tenzen-y) [inline pkg/controller/admissionchecks/multikueue/testdata/worker1KubeConfig:1]: Uhm, I don't like having such a test.
Should we generate this file in `bin` or `tmp` dir, then can we use the generated testdata?

Comment (trasc) [reply pkg/controller/admissionchecks/multikueue/testdata/worker1KubeConfig:1]: It will make the testing strategy more complicated, (create the temp file, replace the path in the needed clusters in `tc.clusters`, cleanup) . 

Even if probably not used in kueue until now, the `testdata` approach is more or less standard in go,

File: apis/kueue/v1alpha1/multikueue_types.go (hunk 2 above)
Comment (tenzen-y) [reply apis/kueue/v1alpha1/multikueue_types.go:32]: It makes sense. I think that we should mention here that we would suggest using the `SecretLocationType` as much as possible, and the `PathLocationType` is assumed to be used for mounting credentials from nodes via hostPath volumeMount.

However, since this is a non-blocking suggestion, I think that we can work on another PR. I leave on @trasc whether or not we work on it at this PR.

File: pkg/controller/admissionchecks/multikueue/testdata/worker1KubeConfig
Comment (tenzen-y) [reply pkg/controller/admissionchecks/multikueue/testdata/worker1KubeConfig:1]: Uhm, I understand that this is the standard approach, but as much as possible, I want to avoid having such testdata.

However, I don't want to block the merge of this PR due to this discussion. So, we probably can come back here when we have more testdata in the near future.

**Review by tenzen-y — COMMENTED** [review]
/lgtm
/approve
/hold

File: apis/kueue/v1alpha1/multikueue_types.go (hunk 2 above)
Comment (trasc) [reply apis/kueue/v1alpha1/multikueue_types.go:32]: `SecretLocationType` is the default value, we could present come pros/cons when we add the documentation.

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
MultiKueue: Add Path location type for cluster KubeConfigs.
```
